server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -98,6 +126,35 @@ public void startReplication(final SegmentReplicationTarget target) {
         threadPool.generic().execute(new ReplicationRunner(replicationId));
     }
 
+    /**
+     * Checks if checkpoint should be processed
+     *
+     * @param requestCheckpoint       received checkpoint that is checked for processing
+     * @param indexshard      replica shard on which checkpoint is received
+     * @return true if checkpoint should be processed
+     */
+    private boolean shouldProcessCheckpoint(ReplicationCheckpoint requestCheckpoint, IndexShard indexshard) {
+        if (indexshard.getState().equals(IndexShardState.STARTED) == false) {
+            logger.debug("Ignore - shard is not started {}", indexshard.getState());
```

Comment:
nit - Can we be a bit more descriptive in these logs "Ignoring new replication checkpoint - ..."
I think these can also be trace logs?